crates/revm/src/evm_impl.rs

```diff
@@ -167,8 +166,9 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact
                     input: data,
                     gas_limit,
                     context,
+                    is_static: false,
```

Comment:
Isn't this slower than putting at compile time? Can't we have a simpler sig that get us the advantage of this PR without putting staticcheck at runtime?

Reply:
It is, it does not matter a lot, it is just a  few `if` and diff is small